### PR TITLE
PP-637: relay hub gas consumption tests

### PR DESCRIPTION
**Pull request by ironFe93**
## What

Migration of tests from truffle to hardhat/ethers

## Why

Part of the ethers migration project

### CODE REVIEW

File: test/RelayHubGas.test.ts [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+  let relayManager: SignerWithAddress;
+  let relayOwner: SignerWithAddress;
+  let fundedAccount: SignerWithAddress;
+  let relayHubSigner: SignerWithAddress;
+  let owner: Wallet;
+  let provider: providers.JsonRpcProvider;
+  const gasPrice = 1;
+  const gasLimit = 4e6;
+
+  const deployHub = (penalizerAddress: string) => {
```
Comment (antomor) [inline test/RelayHubGas.test.ts:67]: Couldn't we use the same function defined in `RelayHub.test.ts`?

File: test/RelayHubGas.test.ts [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+            cumulativeGasUsed: cumulativeGasUsed.toNumber(),
+            detailedEstimation: detailedEstimation.toNumber(),
+            internalDestinationCallCost,
+            internalTokenCallCost,
+          };
+        };
+        it('gas prediction tests - with token payment', async function () {
+          let message =
+            'RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING ';
+          message = message.concat(message);
```
Comment (antomor) [inline test/RelayHubGas.test.ts:429]:
```suggestion
          const message =
            'RIF ENVELOPING '.repeat(22);
```

File: test/RelayHubGas.test.ts [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+
+        it('gas prediction tests - without token payment', async function () {
+          let message =
+            'RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING RIF ENVELOPING ';
+          message = message.concat(message);
+          message = message.concat(message);
+          message = message.concat(message);
+          message = message.concat(message);
+          message = message.concat(message);
+          message = message.concat(message);
```
Comment (antomor) [inline test/RelayHubGas.test.ts:456]:
```suggestion
          const message =
            'RIF ENVELOPING '.repeat(77);
```

**Review by antomor — APPROVED** [review]
@ironFe93 Thank you for the changes, it's surely a big improvement. Apart from the minor comments I left, wdyt if we join some log lines to reduce them. For instance, instead of having:
```
console.log(
            '--------------- Destination Call Without enveloping------------------------'
          );
          console.log(
            `Gas Used: ${callWithoutRelayReceipt.gasUsed.toNumber()}, Cummulative Gas Used: ${cumulativeGasUsedWithoutRelay}`
          );
```
I'd prefer to have something like:
```
          console.log(
            `Destination Call Without enveloping - Gas Used: ${callWithoutRelayReceipt.gasUsed.toNumber()}, Cummulative Gas Used: ${cumulativeGasUsedWithoutRelay}`
          );
```
Furthermore, I'd avoid printing something like:
```
          console.log('---------------------------------------');
```

File: test/RelayHubGas.test.ts (hunk 2 above)
Comment (ironFe93) [reply test/RelayHubGas.test.ts:429]: oh nice, I didn't know this was a thing

File: test/RelayHubGas.test.ts (hunk 1 above)
Comment (ironFe93) [reply test/RelayHubGas.test.ts]: even better, the deployRelayHub function in testUtils is now working so I'm using that one.

File: test/RelayHubGas.test.ts (hunk 2 above)
Comment (ironFe93) [reply test/RelayHubGas.test.ts]: done

File: test/RelayHubGas.test.ts (hunk 3 above)
Comment (ironFe93) [reply test/RelayHubGas.test.ts]: done

**Comment by ironFe93** [comment]
Reduced the lines of code used to log the estimations

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**